### ISSUE TITLE
Crash in example project with no changes [Android]

### ISSUE THREAD

**Issue by davidpricedev** [evollu/react-native-fcm#961]
I was trying to test out the example project, but it seems to just crash for me.

Steps to reproduce: 

1. clone react-native-fcm
1. create a google-services.json file via firebase console for `com.google.firebase.quickstart.fcm` and copy it into the example's android/app folder.
1. `yarn install`
1. `react-native run-android`
1. App crashes.

`adb logcat` shows the reason to be: 

```

05-29 16:10:52.064  1670  1681 I ActivityManager: Start proc 12667:com.google.firebase.quickstart.fcm/u0a88 for activity com.google.firebase.quickstart.fcm/.MainActivity
05-29 16:10:52.072 12667 12667 I zygote  : Not late-enabling -Xcheck:jni (already on)
05-29 16:10:52.073  1670  1686 E memtrack: Couldn't load memtrack module
05-29 16:10:52.073  1670  1686 W android.os.Debug: failed to get memory consumption info: -1
05-29 16:10:52.083 12667 12667 W zygote  : Unexpected CPU variant for X86 using defaults: x86
05-29 16:10:52.092  1670  1694 W ActivityManager: setHasOverlayUi called on unknown pid: 12589
05-29 16:10:52.119 12667 12667 D FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
05-29 16:10:52.124  1393  1393 D gralloc_ranchu: gralloc_alloc: Creating ashmem region of size 8298496
05-29 16:10:52.130 12667 12667 D FirebaseApp: com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
05-29 16:10:52.137 12667 12667 I FirebaseInitProvider: FirebaseApp initialization successful
05-29 16:10:52.139 12667 12667 V fb-UnpackingSoSource: locked dso store /data/user/0/com.google.firebase.quickstart.fcm/lib-main
05-29 16:10:52.139 12667 12667 I fb-UnpackingSoSource: dso store is up-to-date: /data/user/0/com.google.firebase.quickstart.fcm/lib-main
05-29 16:10:52.140 12667 12667 V fb-UnpackingSoSource: releasing dso store lock for /data/user/0/com.google.firebase.quickstart.fcm/lib-main
05-29 16:10:52.161  2568 12606 E ContentStoreEUAS: Failed to commit the deferred actions
05-29 16:10:52.169 12667 12667 D ReactNative: ReactInstanceManager.ctor()
05-29 16:10:52.170 12667 12684 I FA      : App measurement is starting up, version: 11910
05-29 16:10:52.171 12667 12684 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
05-29 16:10:52.171 12667 12684 I FA      : To enable faster debug mode event logging run:
05-29 16:10:52.171 12667 12684 I FA      :   adb shell setprop debug.firebase.analytics.app com.google.firebase.quickstart.fcm
05-29 16:10:52.179 12667 12667 D NetworkSecurityConfig: No Network Security Config specified, using platform default
05-29 16:10:52.180  2500  2641 D EGL_emulation: eglMakeCurrent: 0xa4346fe0: ver 3 0 (tinfo 0x983f4640)
05-29 16:10:52.181  2568  2719 D EGL_emulation: eglMakeCurrent: 0x985ffca0: ver 3 0 (tinfo 0x987ffe00)
05-29 16:10:52.205 12667 12667 D ReactNative: ReactInstanceManager.createReactContextInBackground()
05-29 16:10:52.205 12667 12667 D ReactNative: ReactInstanceManager.recreateReactContextInBackgroundInner()
05-29 16:10:52.206 12667 12667 D ReactNative: ReactInstanceManager.onJSBundleLoadedFromServer()
05-29 16:10:52.207 12667 12667 D ReactNative: ReactInstanceManager.recreateReactContextInBackground()
05-29 16:10:52.207 12667 12667 D ReactNative: ReactInstanceManager.runCreateReactContextOnNewThread()
05-29 16:10:52.246 12667 12692 W unknown:InspectorPackagerConnection: Couldn't connect to packager, will silently retry
05-29 16:10:52.263 12667 12691 I zygote  : Thread[15,tid=12691,Native,Thread*=0xa4334600,peer=0x12d0c5c0,"Thread-3"] recursive attempt to load library "/data/app/com.google.firebase.quickstart.fcm-hBTnRP9DWxQ8HZ7rE6bUvA==/lib/x86/libfb.so"
05-29 16:10:52.264 12667 12691 D ReactNative: ReactInstanceManager.createReactContext()
05-29 16:10:52.270  1670  1731 W AppOps  : Finishing op nesting under-run: uid 1000 pkg android code 24 time=0 duration=0 nesting=0
05-29 16:10:52.272 12667 12691 I zygote  : Rejecting re-init on previously-failed class java.lang.Class<com.google.android.gms.maps.GoogleMapOptions>: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/common/internal/safeparcel/zza;
05-29 16:10:52.272 12667 12691 I zygote  :   at void com.airbnb.android.react.maps.AirMapManager.<init>(com.facebook.react.bridge.ReactApplicationContext) (AirMapManager.java:54)
...

05-29 16:10:52.272 12667 12691 I zygote  : Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.android.gms.common.internal.safeparcel.zza" on path: DexPathList[[zip file "/data/app/com.google.firebase.quickstart.fcm-hBTnRP9DWxQ8HZ7rE6bUvA==/base.apk"],nativeLibraryDirectories=[/data/app/com.google.firebase.quickstart.fcm-hBTnRP9DWxQ8HZ7rE6bUvA==/lib/x86, /system/fake-libs, /data/app/com.google.firebase.quickstart.fcm-hBTnRP9DWxQ8HZ7rE6bUvA==/base.apk!/lib/x86, /system/lib, /vendor/lib]]
05-29 16:10:52.272 12667 12691 I zygote  :   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:93)
...

05-29 16:10:52.275  1410  2016 E SurfaceFlinger: ro.sf.lcd_density must be defined as a build property
05-29 16:10:52.275 12667 12696 D OpenGLRenderer: HWUI GL Pipeline
05-29 16:10:52.277 12667 12691 E AndroidRuntime: FATAL EXCEPTION: Thread-3
05-29 16:10:52.277 12667 12691 E AndroidRuntime: Process: com.google.firebase.quickstart.fcm, PID: 12667
05-29 16:10:52.277 12667 12691 E AndroidRuntime: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/android/gms/maps/GoogleMapOptions;
05-29 16:10:52.277 12667 12691 E AndroidRuntime: 	at com.airbnb.android.react.maps.AirMapManager.<init>(AirMapManager.java:54)
```

When openning an issue, please include following information for better support

1. What version of RN and react-native-fcm are you running?
    * The example project on the master branch as of commit 240ea46 (react-native-fcm@15.0.1, react-native@0.55.4)
2. What device are you using? (e.g iOS9 emulator, Android 6 device)?
    * android emulator, pixel 2, api 26
3. Is your app running in foreground, background or not running?
    * not running - crashed

**Comment by evollu** [comment]
you might need to store packages for android or update your android repository

**Comment by davidpricedev** [comment]
I just tried on a windows system (first post in this issue was on a mac) that hadn't had android studio installed before.

* install react native dependencies: `choco install jdk8 python2`
* install android studio, pick all the defaults and recommendations
    * I also added android support and google play to the list of sdk tools
    * I ended up with 8.1 Oreo as the only SDK platform and I have a few SDK tools versions: 23.0.1, 25.0.2, 25.0.3, 27.0.3
* set ANDROID_HOME to the sdk location where android studio put things
* Fired up the default android emulator (Nexus 5x Api 27) 
* Followed the steps in my first post to get this project up and running.

I get the same error messages.

**Comment by evollu** [comment]
try to build with android studio?

**Comment by davidpricedev** [comment]
I get the same thing happening in android studio. It can't debug the project - the crash happens too early. I did not update gradle the way android studio prompted me to. 

I was able to solve my real problem (this was just a problem I ran into while trying to replicate the problem in the example app). I'll close this since I seem to be the only one having issues. 

I think if I spent a bit of time digging into the interdependencies between firebase and the maps dependency I could probably sort it out with a few additional google/android dependencies with `{ force = true }` on them (assuming I haven't simply typo'd things).
